daffodil-core/src/main/scala/org/apache/daffodil/core/grammar/primitives/ChoiceCombinator.scala

```diff
@@ -236,7 +236,7 @@ case class ChoiceCombinator(ch: ChoiceTermBase, alternatives: Seq[Gram])
         }
       })
 
-      val dispatchBranchKeyMap = dispatchBranchKeyValueTuples.toMap.mapValues { gram =>
+      val dispatchBranchKeyMap = dispatchBranchKeyValueTuples.toMap.map { case (k, gram) =>
```

Comment:
This feels like a case where the original mapValues was clearer (because I don't have to wonder what k is), and calling `.view.mapValues` and then `.toMap` later would be clearer.

Reply:
Maybe we can use .map (...k -> v.mapped) under the hood, that way we keep .mapValues?